lib/net/pool.js

```diff
@@ -1883,6 +1894,142 @@ class Pool extends EventEmitter {
     }
   }
 
+  /**
+   * Handle peer `getcfilters` packet.
+   * @method
+   * @private
+   * @param {Peer} peer
+   * @param {GetCFiltersPacket} packet
+   */
+
+  async handleGetCFilters(peer, packet) {
+    if (this.options.selfish)
+      return;
+
+    if (this.chain.options.spv)
+      return;
+
+    if (this.chain.options.prune)
+      return;
+
+    if (this.chain.options.index)
+      return;
+
+    // The filter type should be valid
+    if (packet.filterType !== common.FILTERS.BASIC)
+      return;
+
+    const stopHeight = this.chain.getHeight(packet.stopHash);
+    // The stop hash doesn't exist
+    if (stopHeight == null)
+      return;
+
+    if (this.chain.getHeight(packet.stopHash) - packet.startHeight >= 1000)
+      return;
+
+    for (let i = packet.startHeight; i <= stopHeight; i++) {
+      const filter = await this.chain.getFilter(i);
+      if (filter.filter) {
+        peer.sendCFilter(common.FILTERS.BASIC, i, filter.filter);
```

Comment:
Like the previous comment, the filter type shouldn't be hard coded but taken from the request packet right? (by now we've already checked the filter type is valid)